src/components/Slider.tsx

```diff
@@ -0,0 +1,161 @@
+import React from "react";
+import styled from "styled-components";
+import { Text } from "@arconnect/components-rebrand";
+
+interface SliderProps {
+  value: number;
+  onChange: (value: number) => void;
+  min?: number;
+  max?: number;
+  step?: number;
+  showLabels?: boolean;
+  minLabel?: string;
+  maxLabel?: string;
+  disabled?: boolean;
+  className?: string;
+}
+
+export function Slider({
+  value,
+  onChange,
+  min = 0,
+  max = 100,
+  step = 1,
+  showLabels = true,
+  minLabel,
+  maxLabel,
+  disabled = false,
+  className,
+}: SliderProps) {
+  const handleSliderChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    onChange(Number(event.target.value));
+  };
+
+  const percentage = ((value - min) / (max - min)) * 100;
+
+  return (
+    <SliderContainer className={className}>
+      <SliderWrapper>
+        {showLabels && (
+          <SliderLabel position="left">
+            <Text variant="secondary" size="sm" weight="medium" noMargin>
+              {minLabel || min}
+            </Text>
+          </SliderLabel>
+        )}
+        <SliderTrackContainer>
+          <SliderInput
```

Comment:
Using a controlled input for sliders make them sluggish/slow. A better implementation would:

- Use an uncontrolled range input for a crispier response, with an `inputRef` prop so that the parent can read the value when needed.
- Move `value` to a state inside the component, as there's no need to pass values up and down if the parent component doesn't need to do anything with it. Updates to this state should be wrapped in a React transition.
- Consider also throttle calls to `onChange` so that updates to the parent have a cap on how fast/often they propagate.